### PR TITLE
Add amctl agent deploy command

### PR DESCRIPTION
**Pull request by jhivandb**
## Purpose
CLI users can build agents (`amctl agent build create`) but must switch to the platform UI to deploy. This breaks scriptability and forces context-switching for routine work.

## Goals
Add `amctl agent deploy [agent]` — a fire-and-forget command that POSTs `DeployAgent` with a resolved image and a merged env-var set. By default it picks the latest deployable build; `--build-name` overrides. Env vars supplied via `--env KEY=VALUE` are merged with the agent's current configuration; conflicts require interactive confirmation or `--yes`.

## Approach
- Extended the `Prompter` interface with `Confirm(prompt) (bool, error)` for y/N prompts
- Added `clierr.BuildNotDeployable` wire-stable error code
- Implemented four pure helpers (`parseEnvFlag`, `findLowestEnvironment`, `mergeEnv`, `renderConflictTable`) with table-driven unit tests
- Wired `DeployOptions` / `NewDeployCmd` / `runDeploy` following the canonical Options/RunE/runX pattern from `agent/delete.go`
- Added `--build-name` flag with `GetBuildWithResponse` resolution and build-status error paths
- Added `--env` flag (`StringArrayVar` to avoid comma-splitting) with conflict detection, sensitive-key handling, and three confirmation gates (JSON mode, non-TTY, interactive prompt)
- Fixed build status gate to accept real server wire values (`Completed`, `Succeeded`) rather than the OpenAPI spec's unemitted enum (`BuildCompleted`)

## User stories
- As a CLI user, I can deploy a built agent image without leaving the terminal
- As a CI pipeline, I can deploy with `--yes` to skip interactive confirmation
- As a user overriding env vars, I see a clear diff table of conflicts before confirming

## Release note
New `amctl agent deploy [agent]` command: deploys a built agent image to the lowest pipeline environment with merged env vars, conflict confirmation, and `--build-name` / `--env` / `--yes` flags.

## Documentation
N/A — CLI `--help` text is self-documenting; no external doc site for amctl yet.

## Training
N/A — internal CLI tooling, not a platform feature requiring training content.

## Certification
N/A — CLI tooling, no certification exam impact.

## Marketing
N/A — internal developer tooling.

## Automation tests
- Unit tests
  - 22 `TestDeploy_*` test cases covering: latest-build happy path, `--build-name` resolution, build status error paths (`Running`, `Failed`, `BuildInProgress`, `BuildTriggered`, nil imageId, no builds), env merge (new key, conflict + `--yes`, conflict + prompt accept/decline, JSON mode, non-TTY, sensitive key redaction, malformed `--env`), pipeline errors, non-buildable agent passthrough
  - 8 `TestLinePrompter_Confirm` cases + 1 `ConfirmDeletion` regression test
  - 6 `TestParseEnvFlag` + 4 `TestFindLowestEnvironment` + 6 `TestMergeEnv` + 2 `TestRenderConflictTable` pure helper tests
  - All tests use `httptest.NewServer` stub servers matching real API routes — no mocks
- Integration tests
  - Manually verified end-to-end against local agent-manager stack (k3d + OpenBao + agent-manager-service)

## Security checks
- Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
- Ran FindSecurityBugs plugin and verified report? N/A — Go codebase, not Java; `go vet` clean
- Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
N/A — CLI command, no samples needed. Usage examples in `--help` text.

## Related PRs
None.

## Migrations (if applicable)
N/A — no database or configuration migrations.

## Test environment
- macOS Darwin 25.3.0 (arm64)
- Go (version from go.mod)
- k3d local cluster with agent-manager-service, OpenBao

## Learning
- Investigated server-side build status mismatch: the OpenAPI spec advertises `BuildCompleted`/`BuildInProgress`/`BuildTriggered` but the server emits upstream WorkflowRun phases (`Completed`, `Succeeded`, `Running`, `Failed`, `Pending`). The CLI gate was updated to accept the real wire values. A separate server-side fix to align the spec is recommended as follow-up.

## Summary by CodeRabbit

* **New Features**
  * Added `agent deploy` command to deploy built agent images to the lowest pipeline environment, with env var overrides, conflict detection, and `--build-name` option.
  * Interactive confirmation for env conflicts via a new generic confirm prompt (use `--yes` to bypass).
  * Introduced a stable CLI error code for non-deployable builds.

* **Tests**
  * Added comprehensive tests covering deploy flows, env parsing/merging, conflict handling, prompter behavior, and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/wso2/agent-manager/pull/858)

### CODE REVIEW

File: cli/pkg/cmd/agent/deploy.go [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+	b := builds[0]
+	status := ""
+	if b.Status != nil {
+		status = string(*b.Status)
+	}
+	if !isBuildDeployable(status) || b.ImageId == nil || *b.ImageId == "" {
+		return deployableBuild{}, clierr.Newf(clierr.BuildNotDeployable,
+			"build %q not deployable: status=%s", b.BuildName, status)
+	}
+	return deployableBuild{Name: b.BuildName, ImageID: *b.ImageId, Status: status}, nil
```
[the remark this answers is not in the document]
Comment (jhivandb) [reply cli/pkg/cmd/agent/deploy.go:390]: Intended behavior is to fail if the command is invoked without a buildName and the latest build is still progressing. It should check the latest build and fail fast. If someone started a build and isn't being specific about what they want to deploy, we need to deploy the latest only rather then the last completed build.

**Review by RAVEENSR — APPROVED** [review]
